intel-analytics/BigDL
# [New Feature] Calculating Scales

## What changes were proposed in this pull request?
1. Implement Trait 'MklInt8Convertible', which enriches modules with this trait having the API to calculate scales for input/output/weight, preparing data to perform quantize and converting model from FP32 to INT8.

2. Added extra fields information to save and load from protobuf. 
   1) Flag of whether the feature (FP32 to INT8) is enabled or not
   2) Input dimension mask
   3) Output dimension mask
   4) Weight dimension mask
   5) Input Scales
   6) Output Scales
   7) Weight Scales
  
## How was this patch tested?
1. Enable this feature to Linear, Spatial Convolution, Sequential, ConcatTable
2. Construct a testing model for each type of them
3. Select a sample set of data as testing input
4. Run forward on the testing model with testing input to populate model's output
5. Invoke newly added API calcScales from MklInt8COnvertible trait to populate scales
6. Compared the populated scales with expected values
7. Save the updated model using protocol buffer definition
8. Re-load the model
9. Check if the newly added fields exist in the loaded model  

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX
